### ISSUE TITLE
sentry.properties does not exist, please run `sentry-wizard -i cordova`

### ISSUE THREAD

**Issue by hermitdemschoenenleben** [getsentry/sentry-cordova#12]
When trying to deploy an ionic app, I always get this error:
Sentry: sentry.properties does not exist, please run `sentry-wizard -i cordova`

Running `./node_modules/@sentry/wizard/dist/bin.js -i cordova` does not fix the problem, my plugins/sentry-cordova/sentry.properties file looks like this:
```
defaults.url=https://sentry.io/
defaults.org=XXX
defaults.project=XXX
auth.token=XXX
cli.executable=node_modules/@sentry/cli/bin/sentry-cli
```

**Comment by HazAT** [comment]
How are you deploying your ionic app?
Does it work in debug?

**Comment by hermitdemschoenenleben** [comment]
it happens for both, release builds (`ionic cordova build android --release --prod --aot`) and debug builds (`ionic cordova run android -l -s --debug`):


```
[ben@localhost app]$ ionic cordova build android --release --prod --aot
Running app-scripts build: --prod --aot --platform android --target cordova
[21:31:12]  build prod started ... 
[21:31:12]  clean started ... 
[21:31:12]  clean finished in 3 ms 
[21:31:12]  copy started ... 
[21:31:12]  deeplinks started ... 
[21:31:13]  deeplinks finished in 941 ms 
[21:31:13]  ngc started ... 
[21:31:36]  ngc finished in 23.46 s 
[21:31:36]  preprocess started ... 
[21:31:36]  preprocess finished in 1 ms 
[21:31:36]  webpack started ... 
[21:31:37]  copy finished in 25.02 s 
[21:33:04]  webpack finished in 87.41 s 
[21:33:04]  uglify started ... 
[21:33:04]  sass started ... 
Without `from` option PostCSS could generate wrong source map and will not find Browserslist config. Set it to CSS file path or to `undefined` to prevent this warning.
[21:33:06]  sass finished in 2.63 s 
[21:33:06]  cleancss started ... 
[21:33:09]  cleancss finished in 2.84 s 
[21:33:47]  uglify finished in 43.59 s 
[21:33:47]  postprocess started ... 
[21:33:47]  postprocess finished in 32 ms 
[21:33:47]  lint started ... 
[21:33:47]  build prod finished in 155.67 s 
> cordova build android --release
Preparing Firebase on Android
Sentry: sentry.properties does not exist, please run `sentry-wizard -i cordova`
[ERROR] An error occurred while running cordova build android --release (exit code 
        1).

```

**Comment by hermitdemschoenenleben** [comment]
the bug is also reproducible with a newly created ionic app:

```
ionic start sentrytest blank
cd sentrytest
ionic cordova platform add android
npm i --save @sentry/cordova
cordova plugin add @sentry/cordova
./node_modules/@sentry/wizard/dist/bin.js -i cordova
ionic cordova build android --release --prod --aot
```

**Comment by Brro33** [comment]
Try with SENTRY_SKIP_AUTO_RELEASE=true in front.
ex: $ SENTRY_SKIP_AUTO_RELEASE=true ionic cordova run android --livereload.

**Comment by devjoks** [comment]
I have the same error: 

`Sentry: sentry.properties does not exist, please run 'sentry-wizard -i cordova'`

And I try to set the 

> ENTRY_SKIP_AUTO_RELEASE=true 

but still doesn't work. Only happens in android, in iOS I don't have any problems.

**Comment by SpellChucker** [comment]
Same issue here, I just have to copy my `plugins/sentry-cordova/sentry.properties` over to `platforms/android/` for it to work. iOS seems to be okay, but Android is not copying the `sentry.properties` file over.

**Comment by SpellChucker** [comment]
Running the wizard, and selecting Android, it doesn't seem to register the selection:

```
Running Sentry Wizard...
version: 0.8.3 | sentry-cli version: 1.30.0
Sentry Wizard will you help to configure your project
Thank you for using Sentry :)
? What integration do you want to set up? Cordova
? Select the platforms you like to set up: iOS, Android
will not configure android
```

I selected Android and it chose `iOS, Android`, then says `will not configure android`

**Comment by hermitdemschoenenleben** [comment]
@Brro33 thank you, using `SENTRY_SKIP_AUTO_RELEASE=true`, it works for me

**Comment by karlvd** [comment]
The wizard places the sentry.properties file in [approot]/plugins
When building, it looks for sentry.properties in [approot]/platforms/android

Manually moving the file also works as a workaround

Checked on Win10

**Comment by jasekiw** [comment]
SENTRY_SKIP_AUTO_RELEASE=true doesn't fix anything it just skips the process. The problem is just going to happen later when you actually want to do a release. I am running into to same issues. To be honest I am going to just use raven-js instead because this package doesn't seem reliable.

**Comment by mikejpeters** [comment]
I worked around this issue by putting my `sentry.properties`  file in my project root, and then configuring Cordova to copy the file to the android folder by editing my `config.xml`:

```xml
<platform name="android">
        <resource-file src="sentry.properties" target="sentry.properties" />
```

**Comment by jodybrewster** [comment]
I think I'm going to have to abandon this plugin too.  I'd like to skip the whole wizard thing (wizards are scary) and plug these values in myself through our CI process.  Is this possible? Like --properties "./sentry.properties" or maybe like --org xxxxxx --project xxxxxxxx --auth-token xxxxxxxxxxx

**Comment by HazAT** [comment]
OK guys, help me out here, what would be the preferred way of dealing with this?
Have a separate command which you can run to upload the source maps/debug symbols and not do it magically by default "on prepare"?
You can absolutely do everything manually but there are several steps one has to take.
Sidenote: We need the source maps/debug symbols to deliver correct stack traces.

**Comment by HazAT** [comment]
I am not sure but I guess there is no "I am creating a prod build now" hook in Cordova at least that was the intention of doing it like it is right now.
Probably the `after_prepare` hook  (what we use right now) is called too often.

**Comment by jodybrewster** [comment]
Personally I'm just trying out the plugin and its failing on multiple levels, the wizard will sometimes hang on the wizard and you can't press the key up and down.  Started looking for a way to side load the properties and disable the wizard and it wasn't happening.  I finally got the cordova plugin add script to complete past the wizard after a couple of days of trying.  It would have been easier for me to bypass the wizard and inject the properties file on the cordova plugin add.  If the wizard happens to fail you're kinda toast, even if you have a correct sentry.properties file.  It would be nice to skip wizard we'll cp the props to correct dir if all else fails.  I also have multiple developers working on this project so will they have to run the wizard every time they run platform add android?  Sorry this has been giving me a lot of grief the past couple days.

**Comment by mikejpeters** [comment]
I also prefer to create the *sentry.properties* file myself, and to be able to easily share it with other devs on my team or use it in CI. The `after_prepare` hook in this project already warns if the file doesn't exist, can it just optionally run the wizard in that situation?

Secondly I think it would be better if *sentry.properties* could be in the project root with other configuration (this makes it more visible, and also means it's easy to create before the plugin directory even exists). I currently accomplish this by having a cordova hook that copies it to the sentry plugin folder.

The other issue is that running the wizard is the only easy way to modify the iOS project and add the necessary build steps to upload debug symbols. Shouldn't this be a separate cordova hook included in this project? That would also mean if we add the iOS platform later to a cordova project everything should just work (instead of having to re-run the wizard).

I am happy with how the current hook uploads the source maps, personally I don't think that needs to change except to only be run on `cordova build` / `cordova run`.

**Comment by RoopeHakulinen** [comment]
Related to this conversation, it also seems that it is currently not possible to use _sentry-cordova_ with PhoneGap Build as it is building the plugins only based only on `config.xml`. In my opinion, ideally the configuration file should be located under version control (so not in `plugins/`). E.g. in the root of the app project. 

Another option would be to utilize the plugin variables, like:

```
<plugin name="cordova-plugin-bluetooth-peripheral-usage-description" spec="1.0.0">
    <variable name="TEXT" value="..."/>
</plugin>
```

**Comment by ArnaudSpanneut** [comment]
Should be great to put the configutation in the `config.xml` file.
I have an automatic way to install my plugins for build on platforms like on Circle CI and Buddybuild.
I try to avoid the using of wizard but it's hard to find how to do. Maybe a doc could be helpful :)
I would love to use the sentry cordova plugin to have a better stacktrace

**Comment by dfibuch** [comment]
Any official word on support for PhoneGap Build? Doesn't look like there is any way to get it to work, as mentioned above, so want to know if support is coming or if its not planned at all?

**Comment by HazAT** [comment]
Sorry, it's been so quiet around this issue lately but this is still on our radar.
In general we won't put the config that `sentry-cli` needs into `config.xml` since this would require use to make `sentry-cli` compatible the Cordova `config.xml`. 

I think the solution would be:
1. Move `sentry.properties` to the root folder of the project
2. Run the wizard only when necessary, run on `platform add/remove` instead `plugin add/remove` 

Regarding PhoneGap, I haven't tried this in very long time but will try to make it work. 
But not sure if it ever will work if you use the "PhoneGap Build" cloud build solution.
Since all the stuff regarding the `wizard` and `sentry-cli` is basically only necessary to get the source maps and debug symbols so we can symbolicate your stack traces.

**Comment by dfibuch** [comment]
Thanks for the response. Those 2 points you mentioned will definitely help with Cordova projects and I guess we can then see if it works on PhoneGap build if we provide it the `sentry.properties` file without it having to run the wizard to generate one.

Do you have any idea of timescale on this?

**Comment by RaulTsc** [comment]
That's awesome! Let me know if you need anything from me.

**Comment by HazAT** [comment]
ETA for this is early next week, please stay tuned and use the known workarounds 😅

**Comment by RaulTsc** [comment]
Ok, looking forward to it!

Thanks @HazAT

**Comment by HazAT** [comment]
OK, version `0.10.0` is on it's way.
Stuff I've changed:
- `sentry.properties` now lives in the project root
- sentry-wizard will run only once (not separately for ios/android)
- Updated the hooks to run only on `cordova build`
- Add some additional checks to not let the iOS build fail

I advise you to do a `cordova plugin rm sentry-cordova` and `cordova plugin add sentry-cordova`
to get all the latest stuff.

Also if you are still having troubles with anything, please create a new Ticket since this is already way to large.

**Comment by RaulTsc** [comment]
Awesome!
